src/DependencyInjection/Container.php

```diff
@@ -228,6 +230,27 @@ public function has($id) {
   /**
    * {@inheritdoc}
    */
+  public function createInstance($plugin_id, $configuration) {
+    $service_definition = $this->getDefinition($plugin_id);
+    $service_definition += array(
+      'arguments' => array()
+    );
+
+    array_unshift($service_definition['arguments'], $configuration);
```

Comment:
Lets move all of this into the caller (ContainerAwarePluginManager::createInstance)